File: meal-service/src/main/java/org/ilya/entities/Day.java
package org.ilya.entities;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.List;

@Data
@NoArgsConstructor
@Entity
@Table(name = "days", uniqueConstraints = @UniqueConstraint(columnNames = {"date", "user_id"}))
public class Day {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(unique = true, nullable = false)
    private LocalDate date;

    @OneToMany(mappedBy = "day", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private List<Meal> meals;

}

File: front-service/src/app/models/day.ts
import {Meal} from "./meal";

export interface Day {
  id: number;       // ID дня
  userId: number;   // ID пользователя
  date: string;     // Дата дня в формате ISO 8601 (например, "2024-12-18")
  meals: Meal[];    // Список приемов пищи (завтрак, обед, ужин)
}
